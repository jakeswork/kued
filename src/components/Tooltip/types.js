import { shape, bool, node } from 'prop-types';

export const propTypes = {
  classes: shape({}).isRequired,
  visible: bool,
  children: node,
};

export const defaultProps = {
  visible: false,
  children: null,
};
