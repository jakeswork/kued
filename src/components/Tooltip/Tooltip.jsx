import React from 'react';
import { CSSTransition } from 'react-transition-group';

import { propTypes, defaultProps } from './types';
import Text from '../Text';

const Tooltip = ({
  classes,
  children,
  visible,
  ...props
}) => (
  <CSSTransition
    timeout={300}
    classNames={{
      enter: classes.tooltipEnter,
      enterActive: classes.tooltipEnterActive,
      exit: classes.tooltipExit,
      exitActive: classes.tooltipExitActive,
    }}
    unmountOnExit
    in={visible}
    {...props}
  >
    <div className={classes.root}>
      <Text caption bold>
        { children }
      </Text>
    </div>
  </CSSTransition>
);

Tooltip.propTypes = propTypes;

Tooltip.defaultProps = defaultProps;

export default Tooltip;
