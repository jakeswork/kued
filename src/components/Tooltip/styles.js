export default {
  root: {
    position: 'absolute',
    top: 40,
    left: 0,
    background: 'rgba(0,0,0,0.8)',
    padding: '8px 16px',
    borderRadius: 4,
    zIndex: 999,
    pointerEvents: 'none',
    userSelect: 'none',
    '&:after': {
      bottom: '100%',
      left: 24,
      border: 'solid transparent',
      content: '" "',
      height: 0,
      width: 0,
      position: 'absolute',
      pointerEvents: 'none',
      borderColor: 'transparent',
      borderBottomColor: 'rgba(0,0,0,0.8)',
      borderWidth: 4,
      marginLeft: -4,
    },
  },
  tooltipEnter: {
    opacity: 0,
    transform: 'scale(0.9)',
  },
  tooltipEnterActive: {
    opacity: 1,
    transform: 'translateY(0)',
    transition: 'all .3s',
  },
  tooltipExit: {
    opacity: 1,
  },
  tooltipExitActive: {
    opacity: 0,
    transform: 'scale(0.9)',
    transition: 'all .3s',
  },
};
