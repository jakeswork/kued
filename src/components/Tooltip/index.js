import withStyles from 'react-jss';

import Tooltip from './Tooltip';
import styles from './styles';

export default withStyles(styles)(Tooltip);
