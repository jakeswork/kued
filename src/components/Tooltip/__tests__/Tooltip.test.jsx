import React from 'react';
import { shallow } from 'enzyme';

import Tooltip from '../Tooltip';

const defaultProps = {
  classes: {},
};

describe('The Tooltip instance', () => {
  describe('When called with default props', () => {
    it('should render without throwing errors', () => {
      const wrapper = shallow(<Tooltip {...defaultProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
