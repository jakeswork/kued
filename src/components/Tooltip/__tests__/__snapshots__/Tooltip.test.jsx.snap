// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Tooltip instance When called with default props should render without throwing errors 1`] = `
<CSSTransition
  classNames={
    Object {
      "enter": undefined,
      "enterActive": undefined,
      "exit": undefined,
      "exitActive": undefined,
    }
  }
  in={false}
  timeout={300}
  unmountOnExit={true}
>
  <div>
    <Jss(Text)
      bold={true}
      caption={true}
      h1={false}
      h2={false}
      h3={false}
      h4={false}
      h5={false}
      link={null}
      style={Object {}}
    />
  </div>
</CSSTransition>
`;
